pkg/artifactory/resource_artifactory_ldap_group_setting.go

```diff
@@ -0,0 +1,243 @@
+			},
+		},
+	}
+}
+
+func resourceLdapGroupSettingsRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*resty.Client)
+	ldapGroupConfigs := &XmlLdapGroupConfig{}
+	ldapGroupSetting := unpackLdapGroupSetting(d)
+
+	resp, err := client.R().Get("artifactory/api/system/configuration")
+	if err != nil {
+		return diag.Errorf("failed to retrieve data from <base_url>/artifactory/api/system/configuration during Read")
+	}
+
+	err = xml.Unmarshal(resp.Body(), &ldapGroupConfigs)
+	if err != nil {
+		return diag.Errorf("failed to xml unmarshal ldap group setting during read operation")
+	}
+	matchedLdapGroupSetting := LdapGroupSetting{}
+	for _, iterLdapGroupSetting := range ldapGroupConfigs.LdapGroupSettings.LdapGroupSettingArr {
+		if iterLdapGroupSetting.Name == ldapGroupSetting.Name {
+			matchedLdapGroupSetting = iterLdapGroupSetting
+		}
+	}
+
+	packDiag := packLdapGroupSetting(&matchedLdapGroupSetting, d)
+	if packDiag != nil {
+		return packDiag
+	}
+	return diag.Diagnostics{{
+		Severity: diag.Warning,
+		Summary:  "Usage of Undocumented Artifactory API Endpoints",
+		Detail:   "The artifactory_ldap_group_setting resource uses endpoints that are undocumented and may not work with SaaS environments, or may change without notice.",
+	}}
+}
+
+func resourceLdapGroupSettingsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	unpackedLdapGroupSetting := unpackLdapGroupSetting(d)
+
+	var constructBody = map[string]map[string]map[string]LdapGroupSetting{}
+	constructBody["security"] = map[string]map[string]LdapGroupSetting{}
+	constructBody["security"]["ldapGroupSettings"] = map[string]LdapGroupSetting{}
+	constructBody["security"]["ldapGroupSettings"][unpackedLdapGroupSetting.Name] = *unpackedLdapGroupSetting
+	content, err := yaml.Marshal(&constructBody)
+
+	if err != nil {
+		return diag.Errorf("failed to marshal ldap group settings during Update")
+	}
+
+	err = sendConfigurationPatch(content, m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+	}
+
+	// we should only have one ldap group setting resource, using same id
+	d.SetId(unpackedLdapGroupSetting.Name)
+	return resourceLdapGroupSettingsRead(ctx, d, m)
+}
+
+func resourceLdapGroupSettingsDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapGroupConfigs := &XmlLdapGroupConfig{}
+
+	rsrcLdapGroupSetting := unpackLdapGroupSetting(d)
+
+	response, err := rClient.R().Get("artifactory/api/system/configuration")
+	if err != nil {
+		return diag.Errorf("failed to retrieve data from <base_url>/artifactory/api/system/configuration during Read")
+	}
+	if response.IsError() {
+		return diag.Errorf("Got error response for <base_url>/artifactory/api/system/configuration request during Read")
```

Comment:
what's this `<base_url>` all about? If you're trying to html/xml tag it, then you need a closing brace. If you want a substitution, then the resty client has this value and you can `fmt.Sprintf` it in

Reply:
Removed the base_url text.